src/istgt_iscsi.c

```diff
@@ -6019,7 +6041,7 @@ worker(void *arg)
 	conn->exec_lu_task = NULL;
 	lu_task = NULL;
 
-	pthread_cleanup_push(worker_cleanup, conn);
+//	pthread_cleanup_push(worker_cleanup, conn);
```

Comment:
Why is this needed?

Reply:
Explained above the intent of the code, and the reason for removing it.

Reply:
Thanks, so this PR addresses more things than just the discovery bug. I can't comment on the need or correctness of those issues as I do not know what they are.

Reply:
These changes are related to this issue very remotely as this code uses lot of locks and CAN cause delay in SendTargets processing if this buggy code got hit.